### PR TITLE
update rc/fish.kak

### PR DESCRIPTION
**Pull request by alexherbo2**

### CODE REVIEW

File: rc/base/fish.kak [hunk 1]
```diff
@@ -34,29 +34,42 @@ addhl -group /fish/code regex \b(and|begin|bg|bind|block|break|breakpoint|builti
 # ‾‾‾‾‾‾‾‾
 
 def -hidden _fish_filter_around_selections %{
-    # remove trailing white spaces
-    try %{ exec -draft -itersel <a-x> s \h+$ <ret> d }
+    eval -draft -itersel %{
+        exec <a-x>
```
Comment (mawww) [inline rc/base/fish.kak]: Why do that in multiple execs instead of the current method ?

File: rc/base/fish.kak [hunk 2]
```diff
@@ -34,29 +34,42 @@ addhl -group /fish/code regex \b(and|begin|bg|bind|block|break|breakpoint|builti
...
+        try %{ exec -draft <a-x> <a-k> ^ \h * (end)  $ <ret> <a-\;> <a-?> ^ \h * (begin|for|function|if|switch|while) <ret> s \A | \Z <ret> \' <a-&>                   }
+        try %{ exec -draft <a-x> <a-k> ^ \h * (case) $ <ret> <a-\;> <a-?> ^ \h * (switch)                             <ret> s \A | \Z <ret> \' <a-&> \' <space> <a-gt> }
     }
 }
 
 def -hidden _fish_indent_on_new_line %{
     eval -draft -itersel %{
         # preserve previous line indent
-        try %{ exec -draft <space> K <a-&> }
+        try %{ exec -draft K <a-&> }
```
Comment (mawww) [inline rc/base/fish.kak]: why remove the `<space>` here ? that wont work well if you are appending on a multiline selection

File: rc/base/fish.kak [hunk 3]
```diff
@@ -34,29 +34,42 @@ addhl -group /fish/code regex \b(and|begin|bg|bind|block|break|breakpoint|builti
...
         # filter previous line
         try %{ exec -draft k : _fish_filter_around_selections <ret> }
-        # copy '#' comment prefix and following white spaces
-        try %{ exec -draft k x s ^\h*\K#\h* <ret> y j p }
-        # indent after (case|else) commands
-        try %{ exec -draft <space> k x <a-k> (case|else) <ret> j <a-gt> }
-        # indent after (begin|for|function|if|switch|while) commands and add 'end' command
-        try %{ exec -draft <space> k x <a-k> (begin|for|function|(?<!(else)\h+)if|switch|while) <ret> x y p j a end <esc> k <a-gt> }
+        # indent after start structure
+        try %{ exec -draft k x <a-k> ^ \h * (begin|case|else|for|function|if|switch|while) \b <ret> j <a-gt> }
```
Comment (mawww) [inline rc/base/fish.kak]: For the coding style, I prefer the whole regex to be written in one go: `^\h*(begin|case|...)\b` it does not seems more readable the other way, in particular `\h *` seems curious as the `*` applies directly to the `\h`

**Comment by mawww** [comment]
Hello, any progress regarding the comments I made ?

File: rc/base/fish.kak (hunk 1 above)
Comment (mawww) [inline rc/base/fish.kak]: ping

**Comment by lenormf** [comment]
Shouldn't we replace the `indent` hooks in all the files by the improved one that was pushed to `c-family` recently ?

**Pull request merged**